src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -202,6 +202,68 @@ public bool NoDownloadRequested(ConnectNewHeadersResult connectNewHeadersResult)
+            /// Chain header tree setup. Initial chain has 4 headers.
+            /// SetUp:
+            ///                        =8d=9d=10d
+            ///                   6a=7a=8a=9a
+            /// GENESIS=1=2=3=4=5=
+            ///                   6b=7b=8b=9b
+            ///             3c=4c=5c
+            /// </summary>
+            /// <param name="initialChainSize">Initial chain size.</param>
+            /// <param name="cht">ChainHeaderTree.</param>
+            /// <param name="initialChainTip">Initial chain tip.</param>
+            /// <param name="chainATip">out Chain A tip.</param>
+            /// <param name="chainDTip">out Chain D tip.</param>
+            /// <returns>List of BlockHeaders.</returns>
+            internal IEnumerable<BlockHeader> SetupPeersForTest(int initialChainSize, ChainedHeaderTree cht, ChainedHeader initialChainTip, out ChainedHeader chainATip, out ChainedHeader chainDTip)
+            {
+                int peerAExtension = 4;
+                int peerBExtension = 4;
+                int peerCExtension = 3;
+                int peerDExtension = 3;
+
+                List<BlockHeader> initialchainHeaders = this.ChainedHeaderToList(initialChainTip, initialChainSize + 1);
+                chainATip = this.ExtendAChain(peerAExtension, initialChainTip); // i.e. (h1=h2=h3=h4=h5)=6a=7a=8a=9a
+                ChainedHeader chainBTip = this.ExtendAChain(peerBExtension, initialChainTip); // i.e. (h1=h2=h3=h4=h5)=6b=7b=8b=9b
+                ChainedHeader chainCTip = this.ExtendAChain(peerCExtension, initialChainTip.GetAncestor(2)); // i.e. (h1=h2)=3c=4c=5c
+                chainDTip = this.ExtendAChain(peerDExtension, chainATip.GetAncestor(7)); // i.e. ((h1=h2=h3=h4=h5)=6a=7a)=8d=9d=10d
+
+                List<BlockHeader> peerABlockHeaders = this.ChainedHeaderToList(chainATip, chainATip.Height);
+                List<BlockHeader> peerBBlockHeaders = this.ChainedHeaderToList(chainBTip, chainBTip.Height);
+                List<BlockHeader> peerCBlockHeaders = this.ChainedHeaderToList(chainCTip, chainCTip.Height);
+                List<BlockHeader> peerDBlockHeaders = this.ChainedHeaderToList(chainDTip, chainDTip.Height);
+
+                var listOfAllBlockHeaders = new List<BlockHeader>(initialchainHeaders);
+                listOfAllBlockHeaders.AddRange(peerABlockHeaders);
+                listOfAllBlockHeaders.AddRange(peerBBlockHeaders);
+                listOfAllBlockHeaders.AddRange(peerCBlockHeaders);
+                listOfAllBlockHeaders.AddRange(peerDBlockHeaders);
+                IEnumerable<BlockHeader> listOfAllUniqueBlockHeaders = listOfAllBlockHeaders.Distinct();
+
+                cht.ConnectNewHeaders(0, peerABlockHeaders);
+                cht.ConnectNewHeaders(1, peerBBlockHeaders);
+                cht.ConnectNewHeaders(2, peerCBlockHeaders);
+                cht.ConnectNewHeaders(3, peerDBlockHeaders);
+
+                return listOfAllUniqueBlockHeaders;
```

Comment:
We shouldn't need to return this.

Reply:
@Aprogiena based on your comments, maybe you just rewrite whole thing yourself? cause every single thing is not correct for you.